logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/UpdateSupportedProtocolsUseCase.kt

```diff
@@ -0,0 +1,127 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.feature.user
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.StorageFailure
+import com.wire.kalium.logic.data.client.Client
+import com.wire.kalium.logic.data.client.ClientRepository
+import com.wire.kalium.logic.data.featureConfig.FeatureConfigRepository
+import com.wire.kalium.logic.data.featureConfig.MLSMigrationModel
+import com.wire.kalium.logic.data.featureConfig.MLSModel
+import com.wire.kalium.logic.data.featureConfig.Status
+import com.wire.kalium.logic.data.sync.SlowSyncRepository
+import com.wire.kalium.logic.data.sync.SlowSyncStatus
+import com.wire.kalium.logic.data.user.SupportedProtocol
+import com.wire.kalium.logic.data.user.UserRepository
+import com.wire.kalium.logic.feature.mlsmigration.hasMigrationEnded
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.flatMap
+import com.wire.kalium.logic.functional.fold
+import com.wire.kalium.logic.functional.map
+import com.wire.kalium.logic.feature.user.UpdateSupportedProtocolsUseCase.Result
+import com.wire.kalium.logic.kaliumLogger
+import kotlinx.coroutines.flow.first
+import kotlinx.datetime.Instant
+
+/**
+ * Updates the supported protocols of the current user.
+ *
+ * @return [Result.Success] if the supported protocols were updated successfully, [Result.Failure] otherwise.
+ */
+interface UpdateSupportedProtocolsUseCase {
+    suspend operator fun invoke(): Result
+
+    sealed interface Result {
+        object Success : Result
+        data class Failure(val failure: CoreFailure) : Result
+    }
+}
```

Comment:
Is this something that apps should call directly? Shouldn't it be like other periodic checks that run after sync?

Reply:
It something we should be periodic after sync I just haven't hooked it up in this PR.